front/app/components/EsriMap/index.tsx

```diff
@@ -35,10 +38,12 @@ const EsriMap = ({
   maxZoom,
   layers,
   onClick,
+  onHover,
   height,
   width,
   graphics,
   showFullscreenOption,
+  uiElements,
 }: Props) => {
   const { data: appConfig } = useAppConfiguration();
 
```

Comment:
Instead of one big `useEffect`:

- Put things that you only need to create once (`esriMap`, `mapView`) inside of `useState` using an initializer function (see https://stackoverflow.com/questions/60120261/when-to-use-usestate-initial-value-as-function), so that they only execute once
- For other props that might change during runtime (e.g. `layers`, `width`) create separate `useEffect` blocks for them. Make sure to memoize them in the parent if they are not primitive values (e.g. `layers`). Also make sure to clean up things if necessary (depends on the ArcGIS library implementation), e.g. remove old layers before adding new layers
- `onClick` and `onHover`: also need to memoize these in the parent
- ESRI map event listeners: only add if necessary
- ESRI map on hover: debounce

Reply:
Also for clarity, you could consider renaming everything that is 'static'  like `center` to `initialCenter`, to make sure that these are only used to initialize the map (after that, the center is controller by the user by dragging the map around)